src/orders/types.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  TransactionEventFragment,
+  TransactionItemFragment,
+} from "@saleor/graphql";
+
+/** Type of the trasaction event (e.g. CHARGE_SUCCESS -> CHARGE) */
+export type TransactionEventType =
+  | "REFUND"
+  | "CHARGE"
+  | "AUTHORIZATION"
+  | "CANCEL"
+  | "CHARGEBACK"
+  | "AUTHORIZATION_ADJUSTMENT"
+  | "REFUND_REVERSED";
+
+/** Status of the transaction (e.g. CHARGE_SUCCESS -> SUCCESS) */
+export type TransactionEventStatus =
+  | "SUCCESS"
+  | "FAILED"
+  | "PENDING"
+  | "REQUEST"
+  | null;
+
+export interface TransactionMappingResult {
+  type: TransactionEventType;
+  status: TransactionEventStatus;
+}
+
+export type TransactionFakeEvent = Omit<
+  TransactionEventFragment,
+  "type" | "__typename"
+> & {
+  __typename: "TransactionFakeEvent";
+  mappedResult: TransactionMappingResult;
+};
+
+export type FakeTransaction = Omit<
+  TransactionItemFragment,
+  "events" | "__typename"
+> & { __typename: "FakeTransaction" };
```

Comment:
**question**: what's the reasoning behind splitting fake and real transaction events?

Reply:
Fake transaction and events are used for old Payments API and Gift cards API to look exactly like transactions on the order details page.

Obviously, they are not a real transaction, so we need to map data from Payments and Gift cards into Transaction shape. I wanted more flexibility for that mapping - TransactionEvents require many fields (e.g. amount) that those APIs simply don't have. It's not an issue - our component is made so that it can display those events even when some data is missing.

Another reason is that TransactionEvent `type` enum field (e.g. AUTHORIZATION_SUCCESS) holds two pieces of information that we need to get in order to display them: status (e.g. FAILED / SUCCEEDED) and kind of event (e.g. CHARGE / AUTHORIZATION). In fake events, we only pass those mapped result (e.g. FAILED + CHARGE) because some events don't have a 1:1 mapping to `TransactionEvent.type`

Reply:
I've added comments in code to explain better this separation 😅